hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/hadoop/HdfsSources.java

```diff
@@ -36,49 +41,81 @@
  */
 public final class HdfsSources {
 
+    /**
+     * With the new HDFS API, some of the {@link RecordReader}s return the same
+     * key/value objects for each record, for example {@link LineRecordReader}.
+     * The source makes a copy of each object ot emit them to downstream. But
+     * for the readers which creates a new object for each record, the source
+     * can be configured to not copy the objects but to reuse them.
+     * <p>
+     * Also if you are using a mapper function which creates a new object for
+     * each record then it makes sense to set this property to {@code true} to
+     * avoid unnecessary copying.
+     * <p>
+     * The source copies the objects by serializing and de-serializing it. The
+     * objects should be either {@link Writable} or serializable in a way that
+     * Jet instance can serialize/de-serialize.
+     * <p>
+     * Here is how you can configure the source. Default value is {@code false}.
+     * <pre>{@code
+     * Configuration conf = new Configuration();
+     * conf.set(HdfsSources.REUSE_OBJECT, "true");
```

Comment:
this setting name is misleading. we NEVER want to reuse same objects when reading as it will break many things with respect to snapshotting, among others.

Reply:
I would suggest something like "CLONE_ON_READ" with default to set to true. With some record readers this is not necessary and we should document which readers clone or not.